ts/parts/Tooltip.ts

```diff
@@ -1188,202 +1189,281 @@ H.Tooltip.prototype = {
-            headerHeight = 0,
-            headerTop: boolean,
-            tooltipLabel = this.getLabel(),
-            distributionBoxTop = chart.plotTop;
+        const tooltip = this;
+        const {
+            chart: {
+                chartWidth,
+                chartHeight,
+                plotHeight,
+                plotLeft,
+                plotTop,
+                plotWidth,
+                pointer,
+                renderer: ren,
+                scrollablePixelsX = 0,
+                scrollablePixelsY = 0,
+                scrollingContainer: {
+                    scrollLeft,
+                    scrollTop
+                } = { scrollLeft: 0, scrollTop: 0 },
+                styledMode,
+                xAxis: [{ opposite }]
+            },
+            options,
+            options: {
+                distance = 16,
+                positioner
+            }
+        } = tooltip;
+
+        const boundaries = {
+            left: scrollablePixelsX ? plotLeft : 0,
+            right: scrollablePixelsX ?
+                plotLeft + plotWidth - scrollablePixelsX : chartWidth,
+            top: scrollablePixelsY ? plotTop : 0,
+            bottom: scrollablePixelsY ?
+                plotTop + plotHeight - scrollablePixelsY : chartHeight
+        };
+
+        const tooltipLabel = tooltip.getLabel();
+        const headerTop = Boolean(opposite);
+
+        let distributionBoxTop = plotTop;
+        let headerHeight = 0;
+        let rightAlign = false;
+
+        /**
+         * Calculates the anchor position for the tooltip
+         *
+         * @private
+         * @param {Highcharts.Point} point The point related to the tooltip
+         * @return {Array<number>} Returns the anchor's x and y position
+         */
+        function getAnchor(
+            point: Highcharts.Point & { isHeader?: boolean }
+        ): [number, number] {
+            const { isHeader, plotX = 0, plotY = 0, series } = point;
+
+            let anchorX;
+            let anchorY;
+            if (isHeader) {
+                // Set anchorX to plotX
+                anchorX = plotLeft + plotX - scrollLeft;
+                // Set anchorY to center of visible plot area.
+                anchorY = plotTop + (plotHeight - scrollablePixelsY) / 2;
+            } else {
+                const { xAxis, yAxis } = series;
+                // Set anchorX to plotX. Limit to within xAxis.
+                anchorX = xAxis.pos + clamp(plotX, 0, xAxis.len) - scrollLeft;
+                // Set anchorY to plotY. Limit to within yAxis.
+                anchorY = yAxis.pos + clamp(plotY, 0, yAxis.len) - scrollTop;
+            }
+
+            // Limit values to plot area
+            anchorX = clamp(anchorX, boundaries.left, boundaries.right);
+            anchorY = clamp(anchorY, boundaries.top, boundaries.bottom);
+            return [anchorX, anchorY];
+        }
+
+        /**
+         * Calculates the position of the tooltip
```

Comment:
As above, confusing whether this applies to the `Tooltip` instance (`this`) or a partial tooltip for the split tooltip.